### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by haonh**

### CODE REVIEW

File: app/assets/stylesheets/custom.css.scss [hunk 1]
```diff
@@ -96,3 +101,29 @@ footer {
...
+input, textarea, select, .uneditable-input {
+  border: 1px solid #bbb;
+  width: 100%;
+  margin-bottom: 15px;
+  @include box_sizing;
+}
+
+input {
+  height: auto !important;
+}
```
Comment (ThuBM) [inline app/assets/stylesheets/custom.css.scss]: new line

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,10 @@
 class User < ActiveRecord::Base
+	before_save { email.downcase! }
```
Comment (ThuBM) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 3]
```diff
@@ -1,2 +1,10 @@
...
+  validates :name, presence: true, length: {maximum: Settings.name_length}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true,
+                    format: {with: VALID_EMAIL_REGEX},
```
Comment (ThuBM) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 4]
```diff
@@ -1,2 +1,10 @@
...
+                    uniqueness: {case_sensitive: false}
```
Comment (ThuBM) [inline app/models/user.rb]: indent

File: app/assets/stylesheets/custom.css.scss [hunk 5]
```diff
@@ -96,3 +101,30 @@ footer {
...
+#error_explanation {
```
Comment (ThuBM) [inline app/assets/stylesheets/custom.css.scss]: add new line above this line

**Comment by ThuBM** [comment]
OK

File: app/views/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t "error_message" %> <%= pluralize @user.errors.count, t("error") %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: indent

File: app/models/user.rb [hunk 7]
```diff
@@ -1,2 +1,10 @@
 class User < ActiveRecord::Base
+  before_save {email.downcase!}
+  validates :name, presence: true, length: {maximum: Settings.name_length}
```
Comment (dieunb) [inline app/models/user.rb]: add new white line above this

**Pull request merged**